File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        List<Integer> ans = new ArrayList();
        int depth = 0;
        dfs(root,ans,depth);
        return ans;
    }
    public void dfs(TreeNode root, List<Integer> ans, int depth){
        if(root == null){
            return;
        }
        if(ans.size() == depth){
            ans.add(root.val);
        }
        dfs(root.right, ans, depth+1);
        dfs(root.left, ans, depth+1);

    }
    
}

File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        def dfs(root,ans,depth):
            if(root is None) : return
            if(len(ans) == depth): ans.append(root.val)
            dfs(root.right,ans,depth+1)
            dfs(root.left,ans,depth+1)
        ans = []
        depth = 0
        dfs(root,ans,depth)
        return ans
